app/services/user_updater.rb

```diff
@@ -283,6 +291,56 @@ def updated_associated_accounts(associations)
 
   private
 
+  def update_sidebar_tags(tag_names)
+    if tag_names.blank?
+      SidebarSectionLink.where(user: user, linkable_type: 'Tag').delete_all
+    else
+      SidebarSectionLink.transaction do
+        exisiting_tag_ids = SidebarSectionLink.where(user: user, linkable_type: 'Tag').pluck(:linkable_id)
+        new_tag_ids = Tag.where(name: tag_names).pluck(:id)
+
+        to_delete = exisiting_tag_ids - new_tag_ids
+        to_insert = new_tag_ids - exisiting_tag_ids
+
+        attributes = to_insert.map do |tag_id|
+          {
+            linkable_type: 'Tag',
+            linkable_id: tag_id,
+            user_id: user.id
+          }
+        end
+
+        SidebarSectionLink.where(user: user, linkable_type: 'Tag', linkable_id: to_delete).delete_all if to_delete.present?
+        SidebarSectionLink.insert_all(attributes) if attributes.present?
+      end
+    end
+  end
+
+  def update_sidebar_categories(category_ids)
```

Comment:
Actually `update_sidebar_tags` and `update_sidebar_categories` methods are very similar.

Do you think we should extract general method like:
```
update_sidebar_links(type, user, exisiting_ids, new_ids)
```

Reply:
Good point, I've updated the code to reduce the amount of duplication.